test/unit/fiber_channel.cc

```diff
@@ -78,12 +78,103 @@ fiber_channel_get()
 	status = check_plan();
 }
 
+void
+fiber_channel_graceful_close_false(struct fiber_channel *channel)
+{
+	/* Test fiber_channel_graceful_close == false. */
+	char msg_1;
+	char msg_2;
+	char msg_3;
+	void *ptr = NULL;
+
+	ok(fiber_channel_put_timeout(channel, &msg_1, 0) == 0,
+	   "fiber_channel_put(msg_1)");
+
+	ok(fiber_channel_put_timeout(channel, &msg_2, 0) == 0,
+	   "fiber_channel_put(msg_2)");
+
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == &msg_1, "fiber_channel_get(1)");
+
+	fiber_channel_close(channel);
+
+	ok(channel->is_closed, "is_closed");
+	ok(channel->is_destroyed, "is_destroyed");
+
+	ok(fiber_channel_put_timeout(channel, &msg_3, 0) != 0,
+	   "not fiber_channel_put(msg_3)");
+
+	ptr = NULL;
+	fiber_channel_get(channel, &ptr);
+
+	ok(ptr == NULL, "not fiber_channel_get(1)");
+
+	fiber_channel_delete(channel);
+}
+
+void
+fiber_channel_graceful_close_true(struct fiber_channel *channel)
```

Comment:
AFAIS, the test cases differ only in one condition. So it looks worthful to parametrize it instead of copying.

Reply:
I think, it could be more confusing, if the tests are combined.

Reply:
I don't insist. Will leave the discussion unresolved for Igor's glance.

Reply:
Igor applied this proposal.